modules/data.land/R/ic_process.R

```diff
@@ -83,93 +86,163 @@ ic_process <- function(settings, input, dir, overwrite = FALSE){
+                                  new_site = new.site,
+                                  host = host, overwrite = overwrite$putveg)
+    }
+
     
   }
 
   #--------------------------------------------------------------------------------------------------#
   # Fill settings
   if (!is.null(putveg.id)) {
     
+    # extend the inputs list for ensemble members
+    settings_inputs <- lapply(seq_along(settings$run$inputs), function(x) rep(settings$run$inputs[[x]], each = length((putveg.id))))
+
+    # make sure all sublists are grouped and renamed to have unique tags, e.g.:
+    # <id>
+    #   <id1>...</id1>
+    #        ...
+    #   <idN>...</idN>
+    # </id>
+    # <path>
+    #   <path1>...</path1>
+    #          ...
+    #   <pathN>...</pathN>
+    # </path>
+    settings_inputs <- lapply(seq_along(settings_inputs), function(x){
+      tmp.list <- settings_inputs[[x]]
+      sub_names <- names(settings$run$inputs[[x]])
+      names(settings_inputs[[x]]) <- paste0(names(settings_inputs[[x]]), rep(seq_along(putveg.id), length(settings$run$inputs[[x]])))
+      tmp.list  <- lapply(seq_along(sub_names), function(v) return(settings_inputs[[x]][names(tmp.list) == sub_names[v]]))
+      names(tmp.list) <- sub_names
+      if(is.null(tmp.list$path)) tmp.list$path <- list()
+      return(tmp.list)
+    })
     
-    model_file <- db.query(paste("SELECT * from dbfiles where container_id =", putveg.id), con)
-    
-    # now that we don't have multipasses, convert.input only inserts 1st filename
-    # do we want to change it in convert.inputs such that it loops over the dbfile.insert?
-    path_to_settings <- file.path(model_file[["file_path"]], model_file[["file_name"]])
-    settings$run$inputs[[input$output]][['path']] <- path_to_settings
+    names(settings_inputs) <- names(settings$run$inputs)
     
-    # NOTE : THIS BIT IS SENSITIVE TO THE ORDER OF TAGS IN PECAN.XML
-    # this took care of "css" only, others have the same prefix
-    if(input$output == "css"){  
-      settings$run$inputs[["pss"]][['path']]  <- gsub("css","pss", path_to_settings)
-      settings$run$inputs[["site"]][['path']] <- gsub("css","site", path_to_settings)
+    for(i in seq_along(putveg.id)){
       
-      # IF: For now IC workflow is only working for ED and it's the only case for copying to remote
-      # but this copy to remote might need to go out of this if-block and change
+      model_file <- db.query(paste("SELECT * from dbfiles where container_id =", putveg.id[[i]], "and machine_id =", machine$id), con)
       
-      # Copy to remote, update DB and change paths if needed
-      if (settings$host$name != "localhost") {
-        
-        remote_dir <- file.path(settings$host$folder, paste0(input$source, "_site_", str_ns))
+      # now that we don't have multipasses, convert.input only inserts 1st filename
+      # do we want to change it in convert.inputs such that it loops over the dbfile.insert?
```

Comment:
Not following this comment -- wasn't `put.veg.module` called in a loop so that putveg.id is a whole list of input IDs? Each ID should point to it's own filename, not the filename of the first ensemble member.

Reply:
this was an older comment, the comparison ("now that we have...") was with respect to an initial version of this workflow where I was going through the whole IC multiple times separately for css, pss and site. Now we get into IC-workflow for `css` initially but we end up writing `pss` and `site` too. 

convert.input only inserts `css` to the DB, so when we query file name only the one with `.css` extension comes up, everytime we need pss or site we need to swap for `.css` in the filename. but I think at one point we also discussed having only one entry for ED css/pss/site in the DB, so that they always go together.. although never really forced it I think

anyway, that's what this comment was about, otherwise each ensemble member has their own IDs pointing to their filenames as you said